Big int support

nikhilkalige opened the pull request.

nikhilkalige commented:
@CeleritasCelery I am kind of lost on the core emacs GC, Trace logic. Any suggestions on how I can make progress with putting them into the objects space?

CeleritasCelery commented:
Trace is easy because it is a terminal type (it contains no other pointers to GC objects). That means `Trace` can be left blank. The harder part is that `BigInt` [uses a vector](https://docs.rs/num-bigint/0.4.6/src/num_bigint/biguint.rs.html#34-36) internally and doesn't use the allocator trait (almost no crates do). This means that it will have extra memory that we can't put on the GC Heap. I have been trying to think about the best way to handle this. We have this same situation with the hashtable we are using.

In that case we have a [special field](https://github.com/nikhilkalige/rune/blob/feature/big-ints/src/core/gc/context.rs#L41) in the allocator. In the `IntoObject` impl we [add them to the list](https://github.com/nikhilkalige/rune/blob/feature/big-ints/src/core/object/tagged.rs#L502) and then [drop the ones that did not get traced](https://github.com/nikhilkalige/rune/blob/feature/big-ints/src/core/gc/context.rs#L184) after collection.

nikhilkalige commented:
@CeleritasCelery I added the types in multiples locations. It builds, but I am sure that its wrong. CloneIn and IntoObject are incorrect, I also added lisp_integers, but failed to correctly use in the Block.

nikhilkalige commented:
Floor, Ceil are working. I have added unnecessary changes too.

Also, Seeing this weird error, don't know what's going on. When I print the value it looks correct, but later in the  output its truncated to 55 bits.
```
󰗀  ./target/debug/rune -r -n
> (floor -97651715312475760.0 nil)
Converting float to integer: -97651715312475760 -> -97651715312475760
-36028797018963968
```

nikhilkalige commented:
This is ready to merge. Proptests are still broken, mostly due to 
- Need a way to signal errors
- Format floats in a fashion similar to emacs

CeleritasCelery commented:
Thanks! This looks good. I will see if I can update elprop to better handle it. The float formatting definitely needs to be worked on.
